File: Clisoap2/clisoap2.rb
# clieSoap2.rb - SOAP + Traducción en Ruby
require 'sinatra'
require 'net/http'
require 'json'
require 'uri'

set :port, 8002

get '/' do
  content_type :text
  numero = params['n'] || "0"
  
  # Simula el consumo SOAP que retorna "ten" y se procesa mediante API de traducción
  texto_ingles = "ten"
  url = URI("https://translate.googleapis.com/translate_a/single?client=gtx&sl=en&tl=es&dt=t&q=#{URI.encode_www_form_component(texto_ingles)}")
  
  res = Net::HTTP.get(url)
  datos = JSON.parse(res)
  texto_espanol = datos[0][0][0] rescue texto_ingles
  
  "Ruby Translation Engine: #{texto_ingles} -> Traducido a: #{texto_espanol}"
end

File: Clisoap2/clisoap2.cs
// clieSoap2.cs - Cliente SOAP + Traducción en .NET 10
using System.Text.Json;

var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

app.MapGet("/", async (HttpContext context) => {
    string n = context.Request.Query["n"] ?? "0";

    try {
        var client = new ServiceReference1.NumberConversionSoapTypeClient(
            ServiceReference1.NumberConversionSoapTypeClient.EndpointConfiguration.NumberConversionSoap
        );
        var response = await client.NumberToWordsAsync(ulong.Parse(n));
        string textoIngles = response.Body.NumberToWordsResult;

        // Consumo manual de traducción síncrona/asíncrona vía HTTP
        using var httpClient = new HttpClient();
        string urlTrad = $"https://translate.googleapis.com/translate_a/single?client=gtx&sl=en&tl=es&dt=t&q={Uri.EscapeDataString(textoIngles)}";
        var resTrad = await httpClient.GetStringAsync(urlTrad);
        
        using var doc = JsonDocument.Parse(resTrad);
        string textoEspanol = doc.RootElement[0][0][0].GetString() ?? textoIngles;
        
        return textoEspanol;
    }
    catch (Exception ex) {
        return $"Error: {ex.Message}";
    }
});

app.Run("http://localhost:8002");